src/plugins/intel_cpu/src/emitters/snippets/cpu_runtime_configurator.cpp

```diff
@@ -4,30 +4,41 @@
 
 #include "emitters/snippets/cpu_runtime_configurator.hpp"
 
-#include "snippets/utils/utils.hpp"
 #include "snippets/lowered/loop_manager.hpp"
-
+#include "snippets/lowered/pass/init_loops.hpp"
+#include "snippets/utils/utils.hpp"
 
 namespace ov {
 namespace intel_cpu {
 
+const size_t CPURuntimeConfigurator::rank6D = 6;
+
 CPURuntimeConfigurator::CPURuntimeConfigurator() : ov::snippets::RuntimeConfigurator(std::make_shared<CPURuntimeConfig>()) {
 }
 
 void CPURuntimeConfigurator::update(const ov::snippets::lowered::LinearIRPtr& linear_ir) {
+    m_config->master_shape = linear_ir->get_master_shape();
+
+    ov::snippets::RuntimeConfigurator::LoopInfoRuntimeParamsMap initialized_info;
+    auto shapes = extract_shapes();
+    auto layouts = extract_layouts();
+    if (m_optimizer.need_optimize(m_config->master_shape)) {
+        m_optimizer.update(initialized_info, shapes, layouts, m_config, m_in_num);
+        // Note: since tensor_rank was changed by optimizer, need to adjust the updated rank accordingly to CPU specific
+        m_config->tensor_rank = std::max(m_config->tensor_rank, rank6D);
```

Comment:
Please note: this is one of the left cons of current solution: we have to manually handle tensor rank on CPU side (because base class doesn't know the tensor rank should be handled). Maybe, we can introduce virtual method `update_tensor_rank`?
If we do that, and move `get_kernel_executor_table()->update_state(linear_ir);` to base class (it seems like we can do that because the exec table is stored in base class), the overridden `CPURuntimeConfigurator::update` can be shorten to
```
RuntimeConfigurator::update(linear_ir);
if (linear_ir->is_dynamic())
    update_loop_args(linear_ir);
```

@a-sidorova what do you think?

Reply:
Agree, let's try. Thank you for the idea!

Reply:
The CPU configurator cleansing is complete 😄